src/components/Icon/Icon.tsx

```diff
@@ -44,6 +50,22 @@ class Icon extends UIComponent<Extendable<IconProps>, any> {
   static displayName = 'Icon'
 
   static propTypes = {
+    /** Generic animation property for adding CSS animation to the component's root element */
+    animation: PropTypes.oneOfType([
```

Comment:
Let's pull this to a common place.  All UI components will support this prop as well.